src/lib.rs

```diff
@@ -207,31 +259,37 @@ impl Connection {
     ///
     /// Will return `Err` if `vfs` cannot be converted to a C-compatible
     /// string or if the underlying SQLite open call fails.
-    pub async fn open_in_memory_with_flags_and_vfs(
-        flags: OpenFlags,
-        vfs: &str,
-    ) -> rusqlite::Result<Self> {
+    pub async fn open_in_memory_with_flags_and_vfs(flags: OpenFlags, vfs: &str) -> Result<Self> {
         let vfs = vfs.to_owned();
-        start(move || rusqlite::Connection::open_in_memory_with_flags_and_vfs(flags, &vfs)).await
+        start(move || rusqlite::Connection::open_in_memory_with_flags_and_vfs(flags, &vfs))
+            .await
+            .map_err(Error::Rusqlite)
     }
 
     /// Call a function in background thread and get the result
     /// asynchronously.
-    pub async fn call<F, R>(&self, function: F) -> R
+    ///
+    /// # Failure
+    ///
+    /// Will return `Err` if the database connection has been closed.
+    pub async fn call<F, R>(&self, function: F) -> Result<R>
     where
-        F: FnOnce(&mut rusqlite::Connection) -> R + Send + 'static,
+        F: FnOnce(&mut rusqlite::Connection) -> rusqlite::Result<R> + 'static + Send,
```

Comment:
I made the call function return a `rusqlite::Result<R>` as well, so there's no `Result<Result<R, rusqlite::Error>, crate::Error>` double result, which would be rather annoying.